strongbox-data-service/src/main/java/org/carlspring/strongbox/data/server/EmbeddedOrientDbServer.java

```diff
@@ -0,0 +1,141 @@
+
+import org.carlspring.strongbox.resource.ConfigurationResourceResolver;
+
+import javax.annotation.PostConstruct;
+import javax.annotation.PreDestroy;
+import java.util.LinkedList;
+import java.util.List;
+
+import com.orientechnologies.orient.server.OServer;
+import com.orientechnologies.orient.server.OServerMain;
+import com.orientechnologies.orient.server.config.OServerConfiguration;
+import com.orientechnologies.orient.server.config.OServerEntryConfiguration;
+import com.orientechnologies.orient.server.config.OServerNetworkConfiguration;
+import com.orientechnologies.orient.server.config.OServerNetworkListenerConfiguration;
+import com.orientechnologies.orient.server.config.OServerNetworkProtocolConfiguration;
+import com.orientechnologies.orient.server.config.OServerUserConfiguration;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Component;
+
+/**
+ * An embedded configuration of OrientDb server.
+ *
+ * @author Alex Oreshkevich
+ */
+@Component
+public class EmbeddedOrientDbServer
+{
+
+    private static final Logger logger = LoggerFactory.getLogger(EmbeddedOrientDbServer.class);
+
+    @Value("${org.carlspring.strongbox.data.orientdb.host}")
+    String host;
+
+    private OServer server;
+    private OServerConfiguration serverConfiguration;
+
+    @Value("${org.carlspring.strongbox.data.orientdb.user}")
+    String user;
+
+    @Value("${org.carlspring.strongbox.data.orientdb.password}")
+    String password;
+
+    @PostConstruct
+    public void init()
+            throws Exception
+    {
+        server = OServerMain.create();
+        serverConfiguration = new OServerConfiguration();
+
+        // prepare network configuration
+        OServerNetworkConfiguration networkConfiguration = new OServerNetworkConfiguration();
+        serverConfiguration.network = networkConfiguration;
+
+        networkConfiguration.protocols = new LinkedList<>();
+
+        OServerNetworkProtocolConfiguration binaryProtocol = new OServerNetworkProtocolConfiguration();
+        binaryProtocol.name = "binary";
+        binaryProtocol.implementation = "com.orientechnologies.orient.server.network.protocol.binary.ONetworkProtocolBinary";
+
+        networkConfiguration.protocols.add(binaryProtocol);
+
+        networkConfiguration.listeners = new LinkedList<>();
+
+        OServerNetworkListenerConfiguration binaryListener = new OServerNetworkListenerConfiguration();
+        binaryListener.ipAddress = "0.0.0.0";
+        binaryListener.portRange = "2424-2430";
+        binaryListener.protocol = "binary";
+        binaryListener.socket = "default";
+        networkConfiguration.listeners.add(binaryListener);
+
+        // add users (incl system-level root user)
+        List<OServerUserConfiguration> users = new LinkedList<>();
+        users.add(buildUser(user, password, "*"));
+        users.add(buildUser("root", "root", "*"));
```

Comment:
This is not good and paves the way for a future vulnerability. Please, remove it.